frontend/model/notifications/vuexModule.js

```diff
@@ -1,19 +1,31 @@
 'use strict'
 
 import { Vue } from '@common/common.js'
-import type { Notification } from './types.flow.js'
-import './selectors.js'
-import { compareOnTimestamp, isNew, isOlder } from './utils.js'
-import * as keys from './mutationKeys.js'
 import { cloneDeep } from '~/frontend/model/contracts/shared/giLodash.js'
+import * as keys from './mutationKeys.js'
+import './selectors.js'
+import { MAX_AGE_READ, MAX_AGE_UNREAD } from './storageConstants.js'
+import type { Notification } from './types.flow.js'
+import { age, compareOnTimestamp, isNew, isOlder } from './utils.js'
 
 const defaultState = {
   items: [], status: {}
 }
 
 const getters = {
   notifications (state, getters, rootState) {
-    return state.items.map(item => ({ ...item, ...state.status[item.hash] }))
+    return state.items.map(item => {
+      const notification = { ...item, ...state.status[item.hash] }
+      // Notifications older than MAX_AGE_UNREAD are discarded
+      if (age(notification) > MAX_AGE_UNREAD) {
+        return null
+      } else if (!notification.read && age(notification) > MAX_AGE_READ) {
+        // Unread notifications older than MAX_AGE_READ are automatically
+        // marked as read
+        notification.read = true
```

Comment:
Same question here - should these be marked as read in the state as well, not just the getter, or perhaps instead of in the getter?

Reply:
I don't think so, because the state is handled separately by incoming KV events, and those notifications will not be saved in the state.

Reply:
BTW, I think it makes sense to try out these changes apart from discussing them in the abstract. If you haven't done so, I suggest modifying (locally) `storageConstants.js` to set the READ and UNREAD values to a low value (e.g., I used 1 min and 5 min, resp.) and see what the state looks like using, e.g., the Vue extension or calling `sbp('state/vuex/state')`